### ISSUE TITLE
Using has_many with not_in raises exception

### ISSUE THREAD

**Issue by terraplane** [activerecord-hackery/ransack#736]
So what we have:
```
class Item < ActiveRecord::Base
  belongs_to :category
end
```
and
```
class Category < ActiveRecord::Base
  has_many: items  
end
```

Now I am trying to call this line in console:
`Category.search(items_id_not_in: [1]).result`

and get the error:
```
NoMethodError: undefined method `eq' for #<Arel::Nodes::Equality:0x007fd72e091228>
	from ***/lib/ransack/adapters/active_record/context.rb:192:in `build_correlated_subquery'`
```

Same but with `in` predicate works perfectly:
`Category.search(items_id_in: [1]).result`

Any ideas/suggestions?

**Comment by seanfcarroll** [comment]
This issue appears more to be a HOW-TO issue rather than a bug in Ransack. We are in the process of cleaning up the Ransack gem and repo. If this issue is still relevant could you please ask it on StackOverflow?

If you think it is an issue with Ransack, please provide a repo with the issue and / or a pull request with failing tests. If you don't know how to do this, indicate it in the issue and we will look into it eventually.

Thank you

https://github.com/activerecord-hackery/ransack/blob/master/CONTRIBUTING.md

https://github.com/activerecord-hackery/ransack/issues/794

**Comment by ajporterfield** [comment]
I ran into this today.  I was using a slightly older version (2.1) and my issue was resolved by upgrading to the latest (2.3).  Hope this helps anyone else that finds this issue.
